packages/tracing-internal/src/browser/browserTracingIntegration.ts

```diff
@@ -0,0 +1,529 @@
+          if (!options.instrumentNavigation) {
+            return;
+          }
+
+          if (activeSpan) {
+            DEBUG_BUILD && logger.log(`[Tracing] Finishing current transaction with op: ${spanToJSON(activeSpan).op}`);
+            // If there's an open transaction on the scope, we need to finish it before creating an new one.
+            activeSpan.end();
+          }
+          activeSpan = _createRouteTransaction(context);
+        });
+
+        client.on('startPageLoadSpan', (context: StartSpanOptions) => {
+          // We check this inside of the hook handler, so that if a custom instrumentation triggers this,
+          // we don't need to check this option in the instrumentation, but can simply invoke it
+          // without needing to know the options of this integration
+          if (!options.instrumentPageLoad) {
+            return;
+          }
+
+          if (activeSpan) {
+            DEBUG_BUILD && logger.log(`[Tracing] Finishing current transaction with op: ${spanToJSON(activeSpan).op}`);
+            // If there's an open transaction on the scope, we need to finish it before creating an new one.
+            activeSpan.end();
+          }
+          activeSpan = _createRouteTransaction(context);
+        });
+      }
+
+      if (options.instrumentPageLoad && client.emit && shouldUseDefaultPageLoadSpan) {
+        const context: StartSpanOptions = {
+          name: WINDOW.location.pathname,
+          // pageload should always start at timeOrigin (and needs to be in s, not ms)
+          startTimestamp: browserPerformanceTimeOrigin ? browserPerformanceTimeOrigin / 1000 : undefined,
+          op: 'pageload',
+          origin: 'auto.pageload.browser',
+          metadata: { source: 'url' },
+        };
+        startBrowserTracingPageLoadSpan(context);
+      }
+
+      if (options.instrumentNavigation && client.emit) {
+        addHistoryInstrumentationHandler(({ to, from }) => {
+          /**
+           * This early return is there to account for some cases where a navigation transaction starts right after
+           * long-running pageload. We make sure that if `from` is undefined and a valid `startingURL` exists, we don't
+           * create an uneccessary navigation transaction.
+           *
+           * This was hard to duplicate, but this behavior stopped as soon as this fix was applied. This issue might also
+           * only be caused in certain development environments where the usage of a hot module reloader is causing
+           * errors.
+           */
+          if (from === undefined && startingUrl && startingUrl.indexOf(to) !== -1) {
+            startingUrl = undefined;
+            return;
+          }
+
+          if (from !== to) {
+            startingUrl = undefined;
+            // We check this in here again, as a custom instrumentation may have been triggered in the meanwhile
+            if (shouldUseDefaultNavigationSpan) {
```

Comment:
So I see one possible issue here, which is that possibly the custom instrumentation for this may have a race condition with this and could be triggered after the default implementation 🤔 Two things I can think of to fix this:

1. We allow to call `startBrowserTracingNavigationSpan(null)`, which does not start a span but tells the instrumentation that we should opt out of the default (a bit hacky maybe...)
2. We expose a separate utility for this (e.g. `disableBrowserTracingDefaultNavigationHandling()` or something like this - a bit clearer but more exports that need to be everywhere & documented 🤔